api/app/serializers/spree/v2/storefront/cart_serializer.rb

```diff
@@ -12,15 +12,7 @@ class CartSerializer < BaseSerializer
 
         has_many   :line_items
         has_many   :variants
-        has_many   :promotions, id_method_name: :promotion_id do |cart|
-          # we only want to display applied and valid promotions
-          # sometimes Order can have multiple promotions but the promo engine
-          # will only apply those that are more beneficial for the customer
-          # TODO: we should probably move this code out of the serializer
-          promotion_ids = cart.all_adjustments.eligible.nonzero.promotion.map { |a| a.source.promotion_id }.uniq
-
-          cart.order_promotions.where(promotion_id: promotion_ids).uniq(&:promotion_id)
-        end
+        has_many   :promotions, object_method_name: :valid_promotions, id_method_name: :promotion_ids
```

Comment:
@PSzyszkaSpark we need another method `valid_promotion_ids` for that, `promotion_ids` will return also non-eligible but attached promotions

```suggestion
        has_many   :promotions, object_method_name: :valid_promotions, id_method_name: :valid_promotion_ids
```